### PR TITLE
Add "shell" option to config

### PR DESCRIPTION
**Pull request by honza**
This allows you to configure the shell to use when alacritty starts.

cc: #122

### CODE REVIEW

**Comment by jwilm** [comment]
Why wouldn't you just use the passwd entry for this?

File: src/config.rs [hunk 1]
```diff
@@ -878,6 +881,12 @@ impl Config {
             .map(|p| p.as_path())
     }
 
+    pub fn shell(&self) -> Option<&Path> {
+        self.shell
+            .as_ref()
+            .map(|p| p.as_path())
```
Comment (jwilm) [inline src/config.rs]: This should probably be `.map(PathBuf::as_path)`

File: src/config.rs [hunk 2]
```diff
@@ -196,6 +196,8 @@ pub struct Config {
     #[serde(default="default_mouse_bindings")]
     mouse_bindings: Vec<MouseBinding>,
 
+    shell: Option<PathBuf>,
```
Comment (jwilm) [inline src/config.rs]: Please add a doc comment

File: src/tty.rs [hunk 3]
```diff
@@ -202,14 +203,19 @@ fn get_pw_entry(buf: &mut [i8; 1024]) -> Passwd {
 }
 
 /// Exec a shell
-fn execsh() -> ! {
+fn execsh(config: &Config) -> ! {
     let mut buf = [0; 1024];
     let pw = get_pw_entry(&mut buf);
 
+    let shell = match config.shell() {
+        Some(shell) => shell.to_str().unwrap(),
```
Comment (jwilm) [inline src/tty.rs]: Please refactor to avoid the `unwrap()`

**Comment by juggle-tux** [comment]
oh cool would it be possible to also add a cli flag for this like `-e` in urxvt. I love to use `urxvt -e tmux attach` saves me at least 5sec on shell initialization.

File: src/config.rs (hunk 1 above)
Comment (dannydulai) [inline src/config.rs]: How is this different than the config_path case which uses `.map(|p| p.as_path())` as well? (I'm new to Rust, so forgive the unrelated question)

Comment (jwilm) [inline src/config.rs]: The other case should probably be updated as well. The closure here is just redundant is all.

File: src/config.rs (hunk 2 above)
Comment (CarwynNelson) [inline src/config.rs]: surely some tests can be written for this?

File: src/tty.rs (hunk 3 above)
Comment (CarwynNelson) [inline src/tty.rs]: and this file as well - surely this can be unit tested

File: src/config.rs [hunk 4]
```diff
@@ -196,6 +196,9 @@ pub struct Config {
     #[serde(default="default_mouse_bindings")]
     mouse_bindings: Vec<MouseBinding>,
 
+    /// Path to a shell program to ru on startup
```
Comment (Phrohdoh) [inline src/config.rs]: s/ru/run

**Comment by jwilm** [comment]
Looks good; thanks!

**Pull request merged**